File: solutions/2025-12-03/solution.java
// Valid Palindrome
// Difficulty: Easy
// URL: https://leetcode.com/problems/valid-palindrome/

public boolean canPermutePalindrome(String s) {
  Map<Character, Integer> charCount = new HashMap<>();
  for (char c : s.toCharArray()) {
    charCount.put(c, charCount.getOrDefault(c, 0) + 1);
  }
  int oddCount = 0;
  for (int count : charCount.values()) {
    if (count % 2 != 0) {
      oddCount++;
    }
  }
  return oddCount <= 1;
}

File: solutions/2025-12-03/solution.py
# Valid Palindrome
# Difficulty: Easy
# URL: https://leetcode.com/problems/valid-palindrome/

def canPermutePalindrome(s: str) -> bool:
  char_count = {}
  for char in s:
    char_count[char] = char_count.get(char, 0) + 1
  odd_count = 0
  for count in char_count.values():
    if count % 2 != 0:
      odd_count += 1
  return odd_count <= 1